ibis_ml/steps/_handle_outliers.py

```diff
@@ -142,7 +142,11 @@ def transform_table(self, table: ir.Table) -> ir.Table:
                     (
                         (table[col_name] >= stat["lower_bound"])
                         & (table[col_name] <= stat["upper_bound"])
-                        | (table[col_name].isnull() | table[col_name].isnan())  # noqa: PD003
+                        | (
+                            (table[col_name].isnull() | table[col_name].isnan())  # noqa: PD003
+                            if table[col_name].type().is_floating()
+                            else table[col_name].isnull()  # noqa: PD003
+                        )
```

Comment:
```suggestion
                        | (table[col_name].isnull() | table[col_name].type().is_floating() & table[col_name].isnan()) # noqa: PD003
```
seems much simpler, unless I'm missing something?

Reply:
Yes, it is simpler.